### PR TITLE
feat: Add professional 3D software controls and enhanced gizmos

### PR DESCRIPTION
**Pull request by ToxSam**
Hey! I've added some major UX improvements to make the Creator Hub feel like professional 3D software:

🎮 Camera Controls Overhaul
- Implemented Blender/Maya-style camera controls
- Orbit: Right-click drag OR Alt+Left-click drag
- Pan: Alt+Middle-click drag
- Zoom: Mouse wheel OR Alt+Right-click drag
- Custom pointer handling with proper sensitivity tuning
- Beta clamping to prevent camera flipping

🧭 3D Axis Orientation Gizmo
- Added real-time 3D axis helper in top-right corner
- Color-coded axes (Red/Green/Blue for X/Y/Z)
- Syncs rotation with main scene camera
- 3D rendered cylinders with cone arrows and text labels
- Transparent background overlay (120x120px)

📐 Position Gizmo - Planar Movement
- Enabled plane constraints for 2D movement
- XY, XZ, and YZ plane handles between axes
- Made gizmo 20% larger for better visibility
- Planes scaled to 37.5% and positioned 0.25 units from center
- Smart repositioning system with render observers

🔳 Scale Gizmo - Dual-Axis Scaling
- Created custom thin cube planes (Babylon.js native support)
- XY (light blue), XZ (light green), YZ (light red) planes
- Paper-thin cubes (0.001 thickness) with translucent materials
- Yellow highlight on hover matching standard gizmo behavior
- Custom PointerDragBehavior for dual-axis sensitivity (3.5x/2.5x)
- Proper material management and cleanup

🎯 Focus on Selected Objects
- Press F to focus camera on selected entity
- Auto-calculates bounding box and frames object perfectly
- Smooth camera animation to target

⌨️ Updated UI
- Refreshed shortcuts panel with new camera controls
- Platform-aware key labels (⌘ on Mac, Ctrl on Windows)
- Vertical layout for multi-option shortcuts
- Added Focus on Selected (F key) to shortcuts menu

🛠️ Code Quality
- Added TransformUtils for coordinate space conversions
- Proper observer cleanup and state management
- Render observers with max attempt limits (60 frames)
- Efficient mesh disposal and cleanup

Files changed:
- New: components/Renderer/AxisHelper/ (AxisHelper.tsx, AxisHelper.css)
- Modified: camera.ts, PositionGizmo.ts, ScaleGizmo.ts, GizmoManager.ts
- Updated: Shortcuts.tsx, Renderer.tsx, Gizmos.tsx

Total: ~2000+ lines across 10+ files
Result: Professional-grade 3D editing experience! 🚀

# PR Title

## Context and Problem Statement

[Describe the issue or background that led to this change. What problem are we solving?]

## Solution

[Explain the solution implemented, focusing on the approach and key technical decisions]

Key changes:
- [Bullet points highlighting the main changes]
- [Focus on what changed, not how it changed]
- [Include architectural decisions if relevant]

## Testing

- [ ] [Specific test case 1]
- [ ] [Specific test case 2]
- [ ] [Edge cases tested]
- [ ] [Regression testing performed]

## Impact

[Describe the user-facing impact and any potential side effects or considerations]

## Screenshots

[Include before and after screenshots or videos if it's a UI feature/fix]

### CODE REVIEW

File: packages/inspector/src/components/Renderer/AxisHelper/AxisHelper.tsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
+import React, { useEffect, useRef } from 'react';
+import * as BABYLON from '@babylonjs/core';
+import { useSdk } from '../../../hooks/sdk/useSdk';
+import './AxisHelper.css';
+
+const AxisHelper: React.FC = () => {
```
Comment (cyaiox) [inline packages/inspector/src/components/Renderer/AxisHelper/AxisHelper.tsx]: could you use the [withSdk HoC](https://github.com/decentraland/creator-hub/blob/main/packages/inspector/src/hoc/withSdk.tsx) for this component?

File: packages/inspector/src/lib/babylon/decentraland/GizmoManager.ts [hunk 2]
```diff
@@ -368,6 +368,27 @@ export function createGizmoManager(context: SceneContext) {
             currentTransformer.enable();
           }
 
+          // Disable the camera gizmo (white circle) - only use X/Y/Z axes
+          // Must be done AFTER enabling to ensure it exists
+          if (gizmoManager.gizmos.rotationGizmo) {
+            const rotGizmo = gizmoManager.gizmos.rotationGizmo as any;
```
Comment (cyaiox) [inline packages/inspector/src/lib/babylon/decentraland/GizmoManager.ts]: is this `any` required?

File: packages/inspector/src/components/Renderer/AxisHelper/AxisHelper.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+  useEffect(() => {
+    if (!sdk || !canvasRef.current) return;
+
+    const canvas = canvasRef.current;
+    const engine = new BABYLON.Engine(canvas, true, {
+      preserveDrawingBuffer: true,
+      stencil: true,
+      antialias: true,
+    });
```
Comment (cyaiox) [inline packages/inspector/src/components/Renderer/AxisHelper/AxisHelper.tsx]: is it required to create a new engine and scene?

what do you think about moving this Babylon logic to a new file: `packages/inspector/src/lib/babylon/setup/axisHelper` similar to: https://github.com/decentraland/creator-hub/blob/main/packages/inspector/src/lib/babylon/setup/setup.ts?

File: packages/inspector/src/lib/babylon/decentraland/gizmos/ScaleGizmo.ts [hunk 4]
```diff
@@ -34,6 +48,490 @@ export class ScaleGizmo implements IGizmoTransformer {
...
+    this.addPlaneDragBehavior(xyPlane, 'XY'); // Scales X and Y
+    this.addPlaneDragBehavior(xzPlane, 'XZ'); // Scales X and Z
+    this.addPlaneDragBehavior(yzPlane, 'YZ'); // Scales Y and Z
+    
+    this.planeCubesCreated = true;
+    console.log('[ScaleGizmo] Plane cubes created successfully');
+  }
+  
+  private addPlaneHoverBehavior(planeMesh: any): void {
+    const BABYLON = require('@babylonjs/core');
```
Comment (cyaiox) [inline packages/inspector/src/lib/babylon/decentraland/gizmos/ScaleGizmo.ts]: same here, could we use imports at the top level?

`import { Color3, PointerEventTypes } from '@babylonjs/core'`

File: package-lock.json [hunk 5]
```diff
@@ -529,7 +529,7 @@
     },
     "node_modules/@clack/prompts/node_modules/is-unicode-supported": {
       "version": "1.3.0",
-      "dev": true,
+      "extraneous": true,
```
Comment (cyaiox) [inline package-lock.json]:
```suggestion
      "dev": true,
```

**Comment by cyaiox** [comment]
Hey @ToxSam, overall strunning contribution! We continue evaluating and testing the changes with the product team

**Comment by cyaiox** [comment]
> Hey @cyaiox just added retroactive camera options to keep the old version on a toggle at the seeting, now that version is called free camera
> 
> * Reintroduced legacy FreeCamera with WASD controls alongside ArcRotateCamera
> * Added 'Free Camera Mode' toggle in Preferences panel (defaults to Orbit mode)
> * Implemented hot-switching between camera modes without scene reload
> * Added context-aware gizmo hotkeys (QWER for Orbit, MRXF for Free Camera)
> * Dynamic shortcuts panel updates based on active camera mode
> * Camera resets to default position when switching modes for consistent UX
> * Fixed shortcuts menu text cutoff with concise button labels (RMB/LMB/MMB)
> 
> Addresses feedback from PR #847 review about separating camera implementations
> 
> let me know what you think
> 
>  2025-11-05.20-03-02.mp4

Hey @ToxSam! Stunning work 👏 

mmm, the building flow is failing due a type error: 
```src/lib/data-layer/host/rpc-methods.ts(209,11): error TS2322: Type '() => Promise<InspectorPreferences>' is not assignable to type 'UnaryMethodImplementation<Empty, InspectorPreferencesMessage, DataLayerContext>'.
  Type 'Promise<InspectorPreferences>' is not assignable to type 'Promise<InspectorPreferencesMessage>'.
    Type 'InspectorPreferences' is not assignable to type 'InspectorPreferencesMessage'.
      Types of property 'cameraMode' are incompatible.
        Type 'import("/Users/runner/work/creator-hub/creator-hub/packages/inspector/src/lib/logic/preferences/types").CameraMode' is not assignable to type 'import("/Users/runner/work/creator-hub/creator-hub/packages/inspector/src/lib/data-layer/proto/gen/data-layer.gen").CameraMode'.
          Type '"orbit"' is not assignable to type 'CameraMode'.
Error: Typechecker exited with code 2.
```

could you fix those types errors? and if it's possible, run the inspector tests locally using: `make test-inspector-e2e`?

**Comment by ToxSam** [comment]
Hey @cyaiox  I've fixed the type errors. The issue was that the proto types weren't regenerated after adding CameraMode to the schema.

What I did:
1. Ran make protoc to regenerate TypeScript from proto definitions
2. Added conversion helpers (cameraModeToProto/cameraModeFromProto) between internal string types and proto enums
3. Updated RPC methods and sagas to properly convert between the two type systems

Verification:
- ✅ Typecheck passes - No CameraMode errors
- ⚠️ E2E tests - I tested both my local main branch and a fresh clone. Both fail with different errors (tinypool workers locally, "server not running" on fresh clone). Neither appear related to my type changes. Could you confirm the expected setup for running these tests?

 The type errors that were blocking the build are now completely resolved! Ready for another review
